ejb3/src/main/java/org/jboss/as/ejb3/context/SessionContextImpl.java

```diff
@@ -34,6 +35,14 @@ public class SessionContextImpl extends EJBContextImpl implements SessionContext
     public SessionContextImpl(SessionBeanComponentInstance instance) {
         super(instance);
         stateful = instance.getComponent() instanceof StatefulSessionComponent;
+        // rls debug
+        java.security.Principal p = instance.getComponent().getCallerPrincipal();
+        if (p != null) {
+            ROOT_LOGGER.trace("## SessionContextImpl  principal: " + p.getName()
+                    + ",  this: " + this);
+            //new Throwable("## SessionContextImpl  principal: " + p.getName()
+            //+ ",  this: " + this).printStackTrace();
+        }
```

Comment:
Since this is production code please wrap all this in an 'if (ROOT_LOGGER.isTraceEnabled())' block.

Reply:
Done.
This code WILL be removed once data have been collected.

Reply:
Understood, and thanks! I'm just cautious about production code.